openerp/addons/base/res/ir_property.py

```diff
@@ -91,68 +76,68 @@ def _update_values(self, cr, uid, ids, values):
                         raise ValueError()
                     field_id = prop.fields_id
                 else:
-                    field_id = self.pool.get('ir.model.fields').browse(cr, uid, field_id)
+                    field_id = self.env['ir.model.fields'].browse(field_id)
 
                 value = '%s,%d' % (field_id.relation, value)
 
         values[field] = value
         return values
 
-    def write(self, cr, uid, ids, values, context=None):
-        return super(ir_property, self).write(cr, uid, ids, self._update_values(cr, uid, ids, values), context=context)
+    @api.multi
+    def write(self, values):
+        return super(Property, self).write(self._update_values(values))
 
-    def create(self, cr, uid, values, context=None):
-        return super(ir_property, self).create(cr, uid, self._update_values(cr, uid, None, values), context=context)
+    @api.model
+    def create(self, values):
+        return super(Property, self).create(self._update_values(values))
 
+    @api.v7
     def get_by_record(self, cr, uid, record, context=None):
-        if record.type in ('char', 'text', 'selection'):
-            return record.value_text
-        elif record.type == 'float':
-            return record.value_float
-        elif record.type == 'boolean':
-            return bool(record.value_integer)
-        elif record.type == 'integer':
-            return record.value_integer
-        elif record.type == 'binary':
-            return record.value_binary
-        elif record.type == 'many2one':
-            if not record.value_reference:
+        return self.browse(cr, uid, record.id, context).get_by_record()
```

Comment:
```
property_obj = self.pool['ir.property']
record = property_obj.browse(cr, uid, ids, context=context)
property_obj.get_by_record(cr, SUPERUSER_ID, record, context=context)
```
If we pass `record` itself to new API call then I am afraid we will loose ```cr, uid, context``` values that were used when calling this function from old API.
So, for instance, if executed above code(with say, other than ```admin``` user), the ```api.v7``` version will call the ```api.v8``` with ```record```'s ```uid``` instead of calling it with ```SUPERUSER_ID```